webapp/src/components/testing/TestSteps.vue

```diff
@@ -177,9 +205,60 @@ const visitedSamples = computed<number>(() => {
     }
   }
 });
+
+const confusionMatrix = computed<{labels : Map<number, string>, matrix : number[][]} | undefined>(() => {
+  if (tested.value === undefined) return undefined;
+  const labels = new Set<number>();
+  const mapLabels = new Map<number, string>();
+  
+  // get all the labels
+  switch (props.task.trainingInformation.dataType) {
+    case "image":
+      (tested.value as Tested["image"]).forEach(({ output }) => {
+        labels.add(output.label);
+        labels.add(output.predicted);
+        mapLabels.set(output.label, output.truth);
+      });
+      break;
+    case "text":
+      return undefined;
+    case "tabular":
+      (tested.value as Tested["tabular"]).results.forEach(({ output }) => {
+        labels.add(output.label);
+        labels.add(output.predicted);
+        mapLabels.set(output.label, output.truth);
+      });
+      break;
+    default: {
+      const _: never = props.task.trainingInformation;
+      throw new Error("should never happen");
+    }
+  }
+  const size = Math.max(labels.size, Math.max(...Array.from(labels)));
```

Comment:
the rest expansion operator (`...`) can actually work with any iterable (which `Set` is).

```suggestion
  const size = Math.max(labels.size, ...labels);
```

that being said, better to read the list of existing labels from `Task.LABEL_LIST` as to always have the same size for a given task.